### PR TITLE
Reworking contributions show respond

### PR DESCRIPTION
**Pull request by svileshina**
### Why

In an effort to make things more RESTful where it is simple to do so, this PR combines the show and respond views into show and then uses pundit to differentiate between the views. There are also quite a few places where I then either removed the "Respond" button if both existed, or left it as "Respond" but changed the path. 

Outstanding issues/things I could use some help with:
- [ ] The contributions policy should be more specific I think - all people have permission to view the "show" but it is just what it displays that is being determined by pundit
- [ ] requests/contributions_spec has two failing tests possibly because:  
- [ ] ListBrowser.vue had a showRespondColumn and a "contribution.respond_path" that I need to change to use show

### Pre-Merge Checklist

- [ ] All new features have been described in the pull request
- [ ] Security & accessibility have been considered
- [ ] High quality tests have been added, or an explanation has been given why the features cannot be tested
- [ ] New features have been documented, and the code is understandable and well commented
- [ ] Entry added to CHANGELOG.md if appropriate
- [ ] All outstanding questions and concerns have been resolved
- [ ] Any next steps that seem like good ideas have been created as issues for future discussion & implementation

### What
Initial Respond Page
Screen Shot 2021-02-04 at 3 49 51 PM

Both are now the show but it depends on whether you are logged in or not, and if you are an admin or not
Screen Shot 2021-02-04 at 3 37 09 PM
Screen Shot 2021-02-04 at 3 45 54 PM

### Next Steps
This is part of #840 (Re-enabling registration)

### CODE REVIEW

**Review by exbinary — COMMENTED** [review]
Gave it first pass and it looks great! 👏🏾 .

I didn't dig into implementations of the extracted partials, assuming they were mostly copied over.

Couple comments inline.

File: app/views/contributions/_top_level_card.html.erb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      <div class="block"><p class="subtitle"><%= @contribution.description %></p></div>
+      <p class="control">
+        <% if @contribution.matched? %>
+          <i class="far fa-check-circle has-text-primary is-size-5 has-text-weight-semibold"><span class="ml-1">Claimed</span></i>
+        <% else %>
+          <%= button_to "Claim This #{@contribution.type}", new_contribution_claim_path(@contribution), class: "button is-primary", method: :get %>
+        <% end %>
+      </p>
+      <br>
+      <% if policy(:contributions).read? %>
```
[the remark this answers is not in the document]
Comment (svileshina) [reply app/views/contributions/_top_level_card.html.erb]: So I wasn't sure what the best way to go about this is - on one hand this is all going through show and this is the show view (which is "read" in application policy) but a lot of this is makes sense with "change?"

[the remark this answers is not in the document]
Comment (svileshina) [reply app/views/contributions/_top_level_card.html.erb]: So I wasn't sure what the best way to go about this is - on one hand this is all going through show and this is the show view (which is "read" in application policy) but a lot of this is makes sense with "change?". Also calling it "change" when its just a permission to view something is odd in a few places (like the view details of a person's info).

[the remark this answers is not in the document]
Comment (exbinary) [reply app/views/contributions/_top_level_card.html.erb]: Yeah, i see what you mean.

I think there's a way to think about the action distinctly from the view. The `authorize` in the action checks whether a user can access the requested endpoint; they're already there, so in a way this is about checking their _previous_ action. Even though it's rendered at the end of this action, the view is really about what they can do _next_. From that perspective, i think it's fine, and even expected, that the view might refer to various policy actions.

To the other point you're raising: yeah, it doesn't sound right to use `change?` to guard eg. whether person details are visible. Maybe this is indicating the need for a new, more nuanced check, such as `read_details?` or `read_personal_details?` or some such?

**Review by exbinary — APPROVED** [review]
Woohoo!! 🙌🏾

**Pull request merged**